src/Orchard.Web/Modules/Orchard.Setup/Services/SetupService.cs

```diff
@@ -194,19 +195,39 @@ await scope
                 }
             }
 
-            shellSettings.State = TenantState.Running;
-            _orchardHost.UpdateShellSettings(shellSettings);
             return executionId;
         }
 
-        private async Task<string> CreateTenantDataAsync(IServiceScope scope, SetupContext context, ShellContext shellContext)
+        private async Task<string> CreateTenantDataAsync(IServiceScope scope, 
+            SetupContext context, 
+            ShellContext shellContext)
         {
             var recipeManager = scope.ServiceProvider.GetService<IRecipeManager>();
 
             var recipe = context.Recipe;
             var executionId = await recipeManager.ExecuteAsync(recipe);
 
+            // Once the recipe has finished executing, we need to update the shell state to "Running", so add a recipe step that does exactly that.
```

Comment:
Yeah this was from O1, Its actually not really restarting the shell, its just setting the shell to running so just Activating it, not restarting it.

The restarting of the shell we still need to do.